qlib/Mapper.qm

```diff
@@ -980,8 +1014,8 @@ Mapper mapv(DataMap);
             if (m.code) {
                 try {
                     v = do_list
-                        ? (map m.code($1, getrec(rec, $#)), v)
-                        : m.code(v, rec);
+                        ? (map m.code($1, getrec(rec + crec, $#)), v)
```

Comment:
there are two considerations, performance & compatibility - adding `crec` before calling `mapFieldIntern()` takes care of both